grpc/grpcmetrics/highcardmetrics/server.go

```diff
@@ -0,0 +1,162 @@
+package highcardmetrics
+
+import (
+	"context"
+	"regexp"
+	"strings"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/heroku/x/go-kit/metrics"
+	"github.com/heroku/x/go-kit/metricsregistry"
+	"github.com/heroku/x/grpc/grpcmetrics"
+)
+
+// NewUnaryServerInterceptor returns an interceptor for unary server calls
+// which will report metrics to the given provider.
+func NewUnaryServerInterceptor(p metrics.Provider) grpc.UnaryServerInterceptor {
+	r0 := metricsregistry.New(p)
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (_ interface{}, err error) {
+		r1 := metricsregistry.NewPrefixed(r0, "grpc.server")
+
+		defer func(begin time.Time) {
+			service, method := parseFullMethod(info.FullMethod)
+			labels := []string{"service", service, "method", method, "response-status", code(err)}
+
+			instrumentMethod(r1, labels, time.Since(begin))
+		}(time.Now())
+
+		return handler(ctx, req)
+	}
+}
+
+// NewStreamServerInterceptor returns an interceptor for stream server calls
+// which will report metrics to the given provider.
+func NewStreamServerInterceptor(p metrics.Provider) grpc.StreamServerInterceptor {
+	r0 := metricsregistry.New(p)
+	return func(srv interface{}, ss grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) (err error) {
+		r1 := metricsregistry.NewPrefixed(r0, "grpc.server")
+
+		service, method := parseFullMethod(info.FullMethod)
+
+		labels := []string{"service", service, "method", method}
```

Comment:
Will this `service` conflict (or be confused) with the existing `service*` columns?

Reply:
very true. how do you feel about `grpc.service` and `grpc.method`? Or any other suggestions welcome!

Reply:
It's like you can read my mind, I love those choices! 

In fact, I was wondering about changing ours in the httpmetrics middleware to use `http.method`, `http.path`, `http.response-status`.